content/blog/2022-05-24-MLEM-release.md

```diff
@@ -0,0 +1,278 @@
+---
+title: Package and deploy your models with MLEM
+date: 2022-05-24
+description: >
+  Introducing MLEM -- Machine Learning Engineering Management for Model
+  Deployment and Management.
+descriptionLong: >
+  We’re excited to announce the launch of our latest open source offering,
+  [MLEM](https://mlem.ai)! MLEM is a tool that automatically extracts meta
+  information like environment and frameworks from models and standardizes that
+  information into a human-readable format within Git. ML teams can then use the
+  model information for deployment into downstream production apps and services.
+  MLEM easily connects to solutions like Heroku to dramatically decrease model
+  deployment time.
+picture: 2022-05-24/mlem-rocket.png
+author: aguschin
+# commentsUrl: TODO
+tags:
+  - Machine Learning
+  - Deployment
+  - Model Registry
+  - MLOps
+---
+
+We built MLEM to address issues that MLOps teams have around managing model
+information as they move them from training and development to production and,
+ultimately, retirement. MLEM is meant to help teams automate the collection of
+information around how the model was trained, what the model is for, and
+operational requirements around deployment.
+
+Just like all our [other](https://dvc.org) [tools](https://cml.dev), MLEM uses
+your Git service to store model information and connects with CI/CD solutions
+for deployment (like Heroku). This Git-based model
+([one of our core philosophies](https://iterative.ai/why-iterative/)) aligns
+model operations and deployment with software development teams – information
+and automation is all based on familiar DevOps tools – so that deploying any
+model into production is that much faster.
+
+With MLEM, ML teams get:
+
+- Human-readable information about a model for search and documentation
+- One-step automated deployment across any cloud
+- Fast model registry setup based on Git
+
+# Human-readable models
+
+It's easy to start using MLEM, since it integrates nicely in your existing
+training workflows by adding a couple of lines:
+
+```python
+import mlem
+mlem.api.save(
+    my_model,
+    "mlem-model",
+    sample_data=train
+)
+```
+
+That produces two files: model binary and model metadata, which is a `.mlem`
+file:
+
+```shell
+$ ls models
+mlem-model mlem-model.mlem
+```
+
+MLEM automatically detects everything you need to run the model: ML framework
+you use, as well as model dependencies (i.e. Python requirements), methods and
+input/output data schema. MLEM saves this information in a simple human-readable
+YAML file:
+
+```yaml
+artifacts:
+  data:
+    hash: b7f7e869f2b9270c516b546f09f49cf7
+    size: 166864
+    uri: mlem-model
+description: Random Forest Classifier
+labels:
+  - random-forest
+  - classifier
+model_type:
+  methods:
+    predict_proba:
+      args:
+        - name: data
```

Comment:
I know nothing of this, but is the '-' before 'name' supposed to be there?

Reply:
Yes. It's basically a list (`-` is a list item enumeration) and then `x: y` is a dict. So in Python it would be like
```python
{
  "args":
    [
      {"name": "data", "type_": ...},
      ...
    ]
}
```